simulation/amaru-sim/src/simulator/generate.rs

```diff
@@ -329,12 +319,89 @@ pub fn generate_arrival_times<R: Rng>(
     arrival_times
 }
 
+pub fn generate_entries<R: Rng>(
+    file_path: &PathBuf,
+    start_time: Instant,
+    mean_millis: f64,
+    number_of_upstream_peers: u8,
+) -> impl Fn(&mut R) -> Vec<Reverse<Entry<ChainSyncMessage>>> {
+    move |rng| {
+        let mut entries: Vec<Reverse<Entry<ChainSyncMessage>>> = vec![];
+        for client in 1..=number_of_upstream_peers {
+            let messages =
+                generate_inputs(rng, file_path).expect("Failed to generate inputs from chain file");
+            let arrival_times =
+                generate_arrival_times(rng, start_time, mean_millis, messages.len());
```

Comment:
is this the main change in the PR, allowing arrival times to be close to each other?